### PR TITLE
Add ~/.ssh/config management

### PR DESCRIPTION
**Pull request by roedie**
This adds the ability to manage the ~/.ssh/config file for users.

### CODE REVIEW

File: users/init.sls [hunk 1]
```diff
@@ -208,6 +208,24 @@ users_ssh_auth_delete_{{ name }}_{{ loop.index0 }}:
...
 
+{% if 'ssh_config' in user %}
+users_ssh_config_{{ name }}:
+  file.managed:
+    - name: {{ home }}/.ssh/config
+    - user: {{ name }}
+    - group: {{ user_group }}
+    - mode: 640
+    - contents: |
+        # Managed by Saltstack
```
Comment (iggy) [inline users/init.sls]: + "Do Not Edit"

**Comment by SPoage** [comment]
This makes some pretty broad assumptions about what the user wants their .ssh/config file to look like. Perhaps we should instead implement something wider in scope that allows arbitrary files to be added (via pillar contents or file template) instead?

File: users/init.sls (hunk 1 above)
Comment (roedie) [inline users/init.sls]: Added the note

**Comment by roedie** [comment]
I'm not sure how adding the config as a file or template would make the assumtion better. I always use it in this form and never seen something different. But then again, I do not have all the knowledge in the world ;-)

**Comment by iggy** [comment]
I'll go ahead and merge this. If someone wants to make broader changes later on, they can do that. This is at least a good first step in the right direction (and doesn't impact anything unless the user actually sets up the pillar data).

**Pull request merged**